api/rest/restapi.go

```diff
@@ -669,27 +669,74 @@ func (api *API) allocationHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func match(filters []string, status string) bool {
+	for _, filter := range filters {
+		if status == filter {
+			return true
+		}
+		if filter == "queued" || filter == "error" {
+			if strings.Contains(status, filter) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func globalPinInfosByStatus(filter string, globalPinInfos []types.GlobalPinInfoSerial) []types.GlobalPinInfoSerial {
+	if filter == "" {
+		return globalPinInfos
+	}
+	filters := strings.Split(strings.ToLower(filter), ",")
+
+	var filteredGlobalPinInfos []types.GlobalPinInfoSerial
+
+	for _, globalPinInfo := range globalPinInfos {
+		for _, pinInfo := range globalPinInfo.PeerMap {
+			if match(filters, pinInfo.Status) {
+				filteredGlobalPinInfos = append(filteredGlobalPinInfos, globalPinInfo)
+				break
+			}
+		}
+	}
+
+	return filteredGlobalPinInfos
+}
+
 func (api *API) statusAllHandler(w http.ResponseWriter, r *http.Request) {
 	queryValues := r.URL.Query()
 	local := queryValues.Get("local")
 
+	var globalPinInfos []types.GlobalPinInfoSerial
+
 	if local == "true" {
 		var pinInfos []types.PinInfoSerial
+
 		err := api.rpcClient.Call("",
```

Comment:
While changing this function, can you update `api.rpcClient.Call` to `api.rpcClient.CallContext` and pass in the Request context `r.Context()`.

Reply:
Also can you fix the multilining of the function call so that params aren't on the same line as parens.